File: UserCenter_OCTest/UserCenter_OCTest/GameViewController.m
//
//  GameViewController.m
//  UserCenter_OCTest
//
//  Created by admin on 2021/2/22.
//  Copyright © 2021 os. All rights reserved.
//

#import "GameViewController.h"
#import "IAPViewController.h"
#import "EventViewController.h"

@import UserCenterSDK;

@interface GameViewController ()

@property (nonatomic, strong) UIButton *loginBtn;
@property (nonatomic, strong) UIButton *loginOutBtn;
@property (nonatomic, strong) UIButton *payBtn;
@property (nonatomic, strong) UIButton *eventBtn;

@end

@implementation GameViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor greenColor];
    
    [[UserCenterSDK share] setLoginSuccessCallback:^(NSString * _Nonnull userName, NSString * _Nonnull userId, NSString * _Nonnull access_token) {
        NSLog(@"userName: %@\n userId: %@\n access_token: %@", userName, userId, access_token);
    }];
    
    [[UserCenterSDK share] setLogoutSuccessCallback:^{
        NSLog(@"退出账号成功");
    }];
    
    self.loginBtn = ({
        UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
        btn.frame = CGRectMake([UIScreen mainScreen].bounds.size.width - 300, 100, 50, 30);
        [btn setTitle:@"登录" forState:UIControlStateNormal];
        [btn setBackgroundColor:[UIColor lightGrayColor]];
        [btn addTarget:self action:@selector(loginBtnClicked) forControlEvents:UIControlEventTouchUpInside];
        [self.view addSubview:btn];
        btn;
    });
    self.loginOutBtn = ({
        UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
        btn.frame = CGRectMake([UIScreen mainScreen].bounds.size.width - 300, 320, 100, 30);
        [btn setTitle:@"退出登录" forState:UIControlStateNormal];
        [btn setBackgroundColor:[UIColor lightGrayColor]];
        [btn addTarget:self action:@selector(loginOutBtnClicked) forControlEvents:UIControlEventTouchUpInside];
        [self.view addSubview:btn];
        btn;
    });
    self.payBtn = ({
        UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
        btn.frame = CGRectMake([UIScreen mainScreen].bounds.size.width - 300, 180, 50, 30);
        [btn setTitle:@"支付" forState:UIControlStateNormal];
        [btn setBackgroundColor:[UIColor lightGrayColor]];
        [btn addTarget:self action:@selector(payBtnClicked) forControlEvents:UIControlEventTouchUpInside];
        [self.view addSubview:btn];
        btn;
        });
    self.eventBtn = ({
        UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
        btn.frame = CGRectMake([UIScreen mainScreen].bounds.size.width - 300, 260, 50, 30);
        [btn setTitle:@"事件" forState:UIControlStateNormal];
        [btn setBackgroundColor:[UIColor lightGrayColor]];
        [btn addTarget:self action:@selector(eventBtnClicked) forControlEvents:UIControlEventTouchUpInside];
        [self.view addSubview:btn];
        btn;
    });
}

- (void)loginBtnClicked {
    [[UserCenterSDK share] accountAutoLogin];
}

- (void)loginOutBtnClicked {
    [[UserCenterSDK share]accountLogout];
}

- (void)payBtnClicked {
    [self presentViewController:[[IAPViewController alloc] init] animated:NO completion:nil];
}

- (void)eventBtnClicked {
    [self presentViewController:[[EventViewController alloc] init] animated:NO completion:nil];
}

@end

File: UserCenter_SwiftTest/UserCenter_SwiftTest/GameViewController.swift
//
//  GameViewController.swift
//  UserCenter_SwiftTest
//
//  Created by admin on 2021/2/7.
//  Copyright © 2021 os. All rights reserved.
//

import UIKit
import UserCenterSDK

fileprivate let screenWidth: CGFloat = UIScreen.main.bounds.size.width
fileprivate let screenHeight: CGFloat = UIScreen.main.bounds.size.height

class GameViewController: UIViewController {
    private lazy var loginBtn: UIButton = {
        let btn = UIButton(type: .custom)
        btn.setTitle("登录", for: .normal)
        btn.backgroundColor = .lightGray
        btn.addTarget(self, action: #selector(loginBtnClicked), for: .touchUpInside)
        return btn
    }()
    
    private lazy var iapBtn: UIButton = {
        let btn = UIButton(type: .custom)
        btn.setTitle("内购", for: .normal)
        btn.backgroundColor = .lightGray
        btn.addTarget(self, action: #selector(iapBtnClicked), for: .touchUpInside)
        return btn
    }()
    
    private lazy var eventBtn: UIButton = {
        let btn = UIButton(type: .custom)
        btn.setTitle("事件", for: .normal)
        btn.backgroundColor = .lightGray
        btn.addTarget(self, action: #selector(eventBtnClicked), for: .touchUpInside)
        return btn
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        addChildView()
        view.backgroundColor = .green
        UserCenterSDK.share.loginSuccessCallback = {//登录成功
            (userName, userId, access_token) in
            print("userName: \(userName)")
            print("userId: \(userId)")
            print("access_token: \(access_token)")
        }
        
        UserCenterSDK.share.logoutSuccessCallback = {
            print("退出账号成功")
        }
    }
}

extension GameViewController {
    @objc private func loginBtnClicked() {
        UserCenterSDK.share.accountAutoLogin()
    }
    
    @objc private func iapBtnClicked() {
        self.present(IAPViewController(), animated: false, completion: nil)
    }
    
    @objc private func eventBtnClicked() {
        self.present(EventViewController(), animated: false, completion: nil)
    }
}

extension GameViewController {
    private func addChildView() {
        view.addSubview(loginBtn)
        loginBtn.frame = CGRect(x: screenWidth - 300, y: 100, width: 50, height: 30)
        
        view.addSubview(iapBtn)
        iapBtn.frame = CGRect(x: screenWidth - 300, y: 180, width: 50, height: 30)
        
        view.addSubview(eventBtn)
        eventBtn.frame = CGRect(x: screenWidth - 300, y: 260, width: 50, height: 30)
    }
}
